GAE/src/com/gallatinsystems/gis/map/MapUtils.java

```diff
@@ -174,6 +175,11 @@ public static void recomputeCluster(Cache cache, SurveyedLocale locale, int delt
             }
         }
 
+        // delete locale if the Delta was a subtraction
+        SurveyedLocaleDao slDao = new SurveyedLocaleDao();
+        if (delta < 0) {
```

Comment:
Are we sure we want to delete a `SurveyLocale` on each substraction? I think we need to check for available `SurveyIntance` for this locale?